tests/run/i11332.scala

```diff
@@ -0,0 +1,72 @@
+// scalajs: --skip
+import scala.language.unsafeNulls
+
+import java.lang.invoke._, MethodType.methodType
+
+class Foo:
+  def neg(x: Int): Int = -x
+  def rev(s: String): String = s.reverse
+  def over(l: Long): String = "long"
+  def over(i: Int): String  = "int"
+  def unit(s: String): Unit = ()
+  def obj(s: String): Object = s
+
+object Test:
+  def main(args: Array[String]): Unit =
+    val l = MethodHandles.lookup()
+    val self    = new Foo()
+    val mhNeg   = l.findVirtual(classOf[Foo], "neg", methodType(classOf[Int], classOf[Int]))
+    val mhRev   = l.findVirtual(classOf[Foo], "rev", methodType(classOf[String], classOf[String]))
+    val mhOverL = l.findVirtual(classOf[Foo], "over", methodType(classOf[String], classOf[Long]))
+    val mhOverI = l.findVirtual(classOf[Foo], "over", methodType(classOf[String], classOf[Int]))
+    val mhUnit  = l.findVirtual(classOf[Foo], "unit", methodType(classOf[Unit], classOf[String]))
+    val mhObj   = l.findVirtual(classOf[Foo], "obj", methodType(classOf[Any], classOf[String]))
+    val mhCL    = l.findStatic(classOf[ClassLoader], "getPlatformClassLoader", methodType(classOf[ClassLoader]))
```

Comment:
This broke the CI because some of our jobs are running on Java 8 which does not define this method:
https://github.com/lampepfl/dotty/actions/runs/3524847656/jobs/5910798965

Reply:
https://github.com/lampepfl/dotty/pull/16394